ts/utils/creditCard.ts

```diff
@@ -0,0 +1,109 @@
+import { fromNullable, Option } from "fp-ts/lib/Option";
+import { ImageURISource } from "react-native";
+
+export type IconSource = (string & ImageURISource) | undefined;
+
+type CreditCardDetector = {
+  blocks: Record<string, ReadonlyArray<number>>;
+  re: Record<string, RegExp>;
+  cardIcons: Record<string, IconSource>;
+  getInfo: (pan: string) => Option<string>;
+  getIcon: (pan: Option<string>) => IconSource;
```

Comment:
Since it always returns an image could it be an improvement use only `ImageURISource` as returned type?
```suggestion
  getIcon: (pan: Option<string>) => ImageURISource;
```
It never happens it return a `string` or `undefined`

Reply:
It depends on how general the component need to be. Now "unknown" case is managed with an image mainly for symmetry. It could also be managed returning the string "io-carta" which refers to the credit card icon in IO font. In that case or similar ones we need the type IconSource